android/eir/src/main/java/org/smartregister/fhircore/eir/EirApplication.kt

```diff
@@ -39,6 +42,10 @@ open class EirApplication : Application(), ConfigurableApplication {
 
   override val fhirEngine: FhirEngine by lazy { FhirEngineProvider.getInstance(this) }
 
+  override val authenticatedUserInfo: UserInfo?
+    get() =
+      SharedPreferencesHelper.read(USER_INFO_SHARED_PREFERENCE_KEY, null)?.decodeJson<UserInfo>()
```

Comment:
Why do we need to override this from the base class?

Reply:
Its interface so we need to provide implementation of how app identifies the authenticated user. This would be used across app to pick logged-in user info when needed